src/lib/routes/client-api/feature.ts

```diff
@@ -222,6 +255,57 @@ export default class FeatureController extends Controller {
         }
     }
 
+    async optimal304(
+        req: IAuthRequest,
+        res: Response<ClientFeaturesSchema>,
+    ): Promise<void> {
+        const query = await this.resolveQuery(req);
+
+        const revisionId = await this.eventService.getMaxRevisionId();
+
+        // TODO: We will need to standardize this to be able to implement this a cross languages (Edge in Rust?).
+        const queryHash = hasSum(query);
```

Comment:
Using a standardized hash would probably be the best solution. We use xxh3-128 in Edge. There's a known implementation for node as well, but it currently requires native libraries.

Reply:
Something like: https://www.npmjs.com/package/xxhash however seems to be native node.

Reply:
But it doesn't implement xxh3  128bit though

Reply:
In a perfect world it makes sense to hash in a predictable way. But it also have limits, as it needs to be correct, and puts extra constraints to the  hashing algorithm and how the content is parsed and hashed. E.g. a different order of the query args could yield a different hash. Its not enough to agree on the hash-algorithm itself, but also the payload going in to the hashing algorithm. The payload will be a combination of URL query args, and the environment.

Imho each component can (and should ?) define their own way to query-hash. The edge would be responsible for their clients, and the Unleash API should be responsible for its clients. The edge still gets the config version number, and can use that to determine that the client still has the latest config. 

This is a reversible decision. By implementing this without assumption that different components use the same hashing algorithm, we can still iterate on the solution later, if we see clear benefits of doing that. The other way is more tricky, as we might drive the implementation on a conclusion that it needs to be the same.   

Wdyt @chriswk ?

Reply:
Sounds reasonable. Edge is set up to believe/store the hash that comes from upstream (whether that is another Edge, or an Unleash server) for caching. This means that if you change the URL from an Unleash Server to an Edge Server you'll invalidate the cache, but since we (as of yet) cannot change the Unleash URL at runtime for our SDKs that should be safe.

Reply:
Yes. But I think that's ok for now, the design allows us to normalize the hashing later, if we see extreme value in that